tests/FSharp.Compiler.ComponentTests/ErrorMessages/InvalidLiteralTests.fs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.ErrorMessages
+
+open Xunit
+open FSharp.Test.Utilities
+open FSharp.Compiler.SourceCodeServices
+
+
+module ``Invalid literals`` =
+
+    [<Fact>]
+    let ``Using Active Pattern``() =
+        CompilerAssert.TypeCheckSingleError
```

Comment:
N.B. - not required to get this merged, but we do have a different approach that we're slowly moving tests over to use:

```
FSharp """
// the code
    """
    |> compile
    |> shouldFail
    |> withSingleDiagnostic (Error 123, Line 3, Col 17, Line 3, Col 22, "The message")
    |> ignore

```

Reply:
Ah, I didn't know. I just copied from another test. I'll update!

Reply:
I've updated to:

```fsharp
FSharp """
let (|A|) x = x + 1
let [<Literal>] (A x) = 1
        """
        |> typecheck
        |> shouldFail
        |> withSingleDiagnostic (Error 3391, Line 3, Col 17, Line 3, Col 22, "A [<Literal>] declaration cannot use an active pattern for its identifier")
```

I've changed the `compile` call to `typecheck`, to be in line with what other tests were doing and due to it not making the tests pass due to the code missing a module/namespace. Is that okay?

I also noticed that the other test suites seem to omit the `ignore` call at the end. Do you want me to add it here? It does work without it.

Reply:
The new approach looks a lot nicer BTW 👍

Reply:
I think it's fine without the explicit "ignore".